### PR TITLE
refactor: 将生息演算：沙洲遗闻由任务树重构为任务环，以便使用出口、后续重构 (#10229)

### PR DESCRIPTION
**Pull request by Daydreamer114**
## Continue Working On #10215 

- 此重构结束后会将生息演算所有任务重构为模板任务

## 待测试 | Tests for ALL Clients

- 因原任务大量复用跳过天数的功能，跳过天数部分被重写，需要重点测试
- [x] CN
- [x] EN
- [x] JP
- [x] KR

## 更改项

- 将普通模式写成简单环，去除所有复用
- 去除所有的#self和#back，仅留有一处sub

- 因制造刷点数复用了旧任务，故将部分旧任务保留并以搜索“2.”的方式替换为“2Ex.”
- 部分图片重命名，没有新增图片和OCR

## 待处理 | Help Wanted

- 可以写导航了
- 大量任务需要增加next的内容以保证稳定性
- 针对稳定性的大量测试 (也许直接发内测等issue)

### CODE REVIEW

**Comment by HX3N** [comment]
> 韩文似乎缺少沙中之火的OCR替换

It seems that the previous YostarKR maintainer missed it because they were in the military at the time

**Comment by Daydreamer114** [comment]
> Are you referring to the previous reclamation? 
> 
> Is it necessary to add the parts that were missed in the previous reclamation?

Yeah thank you for the answer.
If "Fire within the Sand" mode won't came back again, there seems no nessecary to finish it.

File: resource/tasks.json [hunk 1]
```diff
@@ -10133,1314 +10133,1507 @@
...
+        "roi": [508, 142, 267, 283]
     },
-    "Reclamation2EnterStage": {
+    "Fire@Reclamation@BuyWaterProcedure": {
         "algorithm": "OcrDetect",
-        "action": "ClickSelf",
-        "text": ["开始", "行动"],
-        "roi": [1077, 499, 203, 146],
-        "next": ["#back"]
+        "text": ["Skip", "建材", "清水", "Skip"]
```
Comment (status102) [inline resource/tasks.json:10246]: ¿¿¿¿¿

Comment (Daydreamer114) [reply resource/tasks.json:10246]: ?

Comment (status102) [reply resource/tasks.json:10246]: 这个Skip是啥

Comment (Daydreamer114) [reply resource/tasks.json:10246]: 不知道啊，问写沙中之火的

**Pull request merged**